drivers/gpio/gpio_dw.c

```diff
@@ -205,28 +303,82 @@ static inline void dw_port_config(struct device *port, int flags)
 static inline int gpio_dw_config(struct device *port, int access_op,
 				 u32_t pin, int flags)
 {
+	const struct gpio_dw_config *config = port->config->config_info;
+	struct gpio_dw_runtime *context = port->driver_data;
+	u32_t port_base_addr = context->base_addr;
+	u32_t data_port = dw_get_data_port(port_base_addr);
+
+	/* Check for invalid pin number */
+	if (pin >= config->bits) {
+		return -EINVAL;
+	}
+
+	/* Does not support disconnected pin */
+	if (flags == GPIO_DISCONNECTED) {
+		return -ENOTSUP;
+	}
+
+	/* Check for an invalid pin configuration */
+	if (((flags & GPIO_INT_ENABLE) != 0U) &&
+	    ((flags & GPIO_INPUT) == 0U)) {
+		/* Interrupt enabled but pin is not set to input */
+		return -EINVAL;
+	}
+
+	/* Only PORT-A support interrupts and de-bounce*/
+	if (data_port != SWPORTA_DR) {
+		if ((flags & GPIO_INT_ENABLE) != 0U) {
+			return -ENOTSUP;
+		}
+
+		if ((flags & GPIO_INT_DEBOUNCE) != 0U) {
```

Comment:
I don't think it's a good practice to ignore the debounce flag if hardware does not support it. Developers would expect this to be working since there is no error. They will be very confused when multiple interrupts are triggered in a very very short period of time for one, say, button down press.

Reply:
Moving this discussion to https://github.com/zephyrproject-rtos/zephyr/issues/19649#issuecomment-539117583